firmware/application/hw/debounce.cpp

```diff
@@ -23,6 +23,31 @@
 
 #include "utility.hpp"
 
+uint8_t Debounce::state() {
+    bool v = !pulse_upon_release_ && (state_ || simulated_pulse_);
+    if (simulated_pulse_)
```

Comment:
Two thoughts: do we need any synchronization with any of this stuff? 
Do you need this test or can you just always set false (like in long press)?

Reply:
Let's put it in for now so I can test with it.

Reply:
Thanks.  It works better than before, but I am looking at eliminating potential synchronization issues as you mentioned.  e.g. There is a possibility that the timer ISR could fire while EventDispatcher is calling the state() function.  I will submit a new PR for subsequent improvements.